### PR TITLE
Modernizing the codebase and minor cleanup

### PR DESCRIPTION
**Pull request by thoov**
cc: @stefanpenner @rwjblue

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -7,23 +7,23 @@ init:
 # Test against these versions of Node.js.
 environment:
   matrix:
-    - nodejs_version: "4"
     - nodejs_version: "6"
+    - nodejs_version: "8"
```
Comment (stefanpenner) [inline appveyor.yml:13]: can you also add `stable`?

**Review by stefanpenner — CHANGES_REQUESTED** [review]
1 nit, then LGTM.

File: package.json [hunk 2]
```diff
@@ -16,15 +16,24 @@
     "index.js"
   ],
   "scripts": {
+    "lint": "eslint *.js test/*.js",
```
Comment (rwjblue) [inline package.json]: Ideally we can use `eslint .`

**Comment by thoov** [comment]
@stefanpenner This should now be ready to merge

**Comment by stefanpenner** [comment]
released as v2.0.0 🎉

**Pull request merged**